### PR TITLE
Updated Developer scripts to run Survey Runner in ECS

### PR DESCRIPTION
**Pull request by jonnyshaw89**
### What is the context of this PR?
Have added the Survey Runner and Survey Launcher deploy modules and updated developer defaults to make them work

Running this PR tests the following PRs
https://github.com/ONSdigital/eq-terraform-ecs/pull/5
https://github.com/ONSdigital/eq-survey-launcher-deploy/pull/1
https://github.com/ONSdigital/eq-survey-runner-deploy/pull/1

### How to review
Run `terraform apply` like normal

### CODE REVIEW

File: developer_defaults.tf [hunk 1]
```diff
@@ -24,73 +24,92 @@ variable "aws_key_pair" {
...
+  default     = "1"
+}
+
+variable "ecs_cluster_max_size" {
+  description = "ECS Cluster Maximum number of instances"
+  default     = "3"
+}
 
 // Survey Runner Application
+variable "survey_runner_s3_secrets_bucket" {
```
Comment (ajmaddaford) [inline developer_defaults.tf]: This variable should be moved next to all the other Survey Runner on ECS specific variables. Also, I think the comment above this line should be `Survey Runner on Elastic Beanstalk`

File: developer_defaults.tf [hunk 2]
```diff
@@ -126,6 +145,27 @@ variable "eb_rolling_update_enabled" {
   default     = "false"
 }
 
+# Survey Runner
```
Comment (ajmaddaford) [inline developer_defaults.tf]: I think this comment should be `Survey Runner on ECS`

File: survey-runner.tf [hunk 3]
```diff
@@ -34,84 +41,142 @@ module "survey-runner-application" {
...
+  database_host           = "${module.survey-runner-database.database_address}"
+  database_port           = "${module.survey-runner-database.database_port}"
+  database_name           = "${var.database_name}"
+  rabbitmq_ip_prime       = "${module.survey-runner-queue.rabbitmq_ip_prime}"
+  rabbitmq_ip_failover    = "${module.survey-runner-queue.rabbitmq_ip_failover}"
+  google_analytics_code   = "${var.google_analytics_code}"
+  survey_runner_min_tasks = "${var.survey_runner_min_tasks}"
+  docker_registry         = "${var.docker_registry}"
+  survey_runner_tag       = "${var.survey_runner_tag}"
+  secrets_file_name       = "${var.secrets_file_name}"
```
Comment (ajmaddaford) [inline survey-runner.tf]: These variables should be namespaced with `survey_runner_` to be consistent with the launcher variables below

File: survey-runner.tf [hunk 4]
```diff
@@ -34,84 +41,142 @@ module "survey-runner-application" {
 }
 
 module "survey-runner-ecs" {
```
Comment (ajmaddaford) [inline survey-runner.tf]: We should rename this something like `eq-ecs`

File: survey-runner.tf [hunk 5]
```diff
@@ -34,84 +41,142 @@ module "survey-runner-application" {
...
+  ecs_cluster_name        = "${module.survey-runner-ecs.ecs_cluster_name}"
+  aws_alb_dns_name        = "${module.survey-runner-ecs.aws_alb_dns_name}"
+  aws_alb_listener_arn    = "${module.survey-runner-ecs.aws_alb_listener_arn}"
+  survey_runner_url       = "https://${var.env}-surveys.${var.dns_zone_name}"
+  s3_secrets_bucket       = "${var.survey_launcher_s3_secrets_bucket}"
+  jwt_signing_key_path    = "${var.survey_launcher_jwt_signing_key_path}"
   jwt_encryption_key_path = "${var.survey_launcher_jwt_encryption_key_path}"
 }
 
+module "survey-launcher-new" {
```
Comment (ajmaddaford) [inline survey-runner.tf]: I think we should use some clearer naming for the modules, something like:
```
survey-runner-on-elastic-beanstalk
survey-runner-on-ecs
survey-launcher-for-elastic-beanstalk
survey-launcher-for-ecs
```

**Review by ajmaddaford — CHANGES_REQUESTED** [review]

**Review by adoublebarrel — APPROVED** [review]
LGTM

**Comment by ajmaddaford** [comment]
All looks good 👍 

I'll leave my blocking review until https://github.com/ONSdigital/eq-survey-launcher-deploy/pull/1 and https://github.com/ONSdigital/eq-terraform-ecs/pull/5 get merged and the module references in this pull request are updated.

**Review by dwyeradam — APPROVED** [review]

**Review by ajmaddaford — APPROVED** [review]

**Pull request merged**